### PR TITLE
(DOCSP-14984): Node.js and React Native landing pages

### PR DESCRIPTION
**Pull request by mongomoe**
### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-14984

### Docs staging link (requires sign-in on MongoDB Corp SSO):
* Node: https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-14984-Node-React-Native-Landing-Page/sdk/node/
* React Native: https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-14984-Node-React-Native-Landing-Page/sdk/react-native/
### Review Guidelines
[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

**Review by jenniferpeshansky — COMMENTED** [review]
overall this is really good and thorough, and I found the code examples really easy to follow and understand

File: examples/node/Examples/landing-page.js [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        name: "Clover",
+        age: 4,
+        type: "Calico",
+      });
+    });
+
+    expect(realm.objects("Cat")[0].name).toBe("Lacy");
+    expect(realm.objects("Cat")[1].name).toBe("Max");
+    expect(realm.objects("Cat")[2].name).toBe("Clover");
+
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply examples/node/Examples/landing-page.js:49]: yup

File: source/examples/generated/node/landing-page.codeblock.node-landing-query-realm-database.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// retrieve the set of Cat objects
+const cats = realm.objects("Cat");
+console.log(`There are ${cats.length} cats`);
+// filter for cats that are older than 7
+const catsOlderThan7 = cats.filtered("age > 7");
+console.log(
+  `There are ${catsOlderThan7.length} cats older than 7 years old`
+);
+// filer for calico cats
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/examples/generated/node/landing-page.codeblock.node-landing-query-realm-database.js:9]: Fixed 👍  (this is just the generated file, but I've re-generated it with the typo fix

File: examples/node/Examples/landing-page.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    realm.write(() => {
+      // update clover's age to 5 years old
+      cloverCat.age = 5;
+    }); // when the transaction completes, the cat's age is updated in the database
+    console.log(`Clover the cat is ${cloverCat.age} years old`);
+    // :code-block-end:
+    expect(cloverCat.age).toBe(5);
+
+    // :code-block-start:  node-landing-watch-for-object-updates
+    // insert a cats into the database and store it in our reference
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply examples/node/Examples/landing-page.js:81]: 👍

File: source/examples/generated/node/landing-page.codeblock.node-landing-sync-data-between-realms.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+var config = {
+  schema: [Cat], // predefined schema
+  sync: {
+    user: app.currentUser,
+    partitionValue: "MyPartitionValue",
+  },
+};
+// Open a realm with a configuration object that with a SyncConfiguration
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/examples/generated/node/landing-page.codeblock.node-landing-sync-data-between-realms.js:8]: Fixed 👍

File: source/examples/generated/node/landing-page.codeblock.node-landing-sync-data-between-realms.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+realm.write(() => {
+  // get a cat from the database to update
+  darukCat = realm.objects("Cat").filtered("name = 'Daruk")[0];
+  // change the cat's name
+  darukCat.name = "Daruk Goron";
+}); // when the transaction completes, the cat's name is updated in the database
+// and synced to the connected Realm App
+
+// use darukCat
+
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/examples/generated/node/landing-page.codeblock.node-landing-sync-data-between-realms.js:24]: Initially, it was meant to mean "use the variable darukCat that is a reference to the cat with the name 'Daruk", but I've removed this comment altogether now.

File: source/examples/generated/node/landing-page.codeblock.node-landing-sync-data-between-realms.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+// start a write transaction
+let darukCat;
+realm.write(() => {
+  // get a cat from the database to update
+  darukCat = realm.objects("Cat").filtered("name = 'Daruk")[0];
+  // change the cat's name
+  darukCat.name = "Daruk Goron";
+}); // when the transaction completes, the cat's name is updated in the database
```
Comment (jenniferpeshansky) [inline source/examples/generated/node/landing-page.codeblock.node-landing-sync-data-between-realms.js:20]: suggestion: should the comment be on the next line?

**Review by jenniferpeshansky — APPROVED** [review]
LGTM with one small suggestion!

File: source/examples/generated/node/landing-page.codeblock.node-landing-sync-data-between-realms.js (hunk 6 above)
Comment (mongomoe) [reply source/examples/generated/node/landing-page.codeblock.node-landing-sync-data-between-realms.js:20]: Personally I like it here, as its similar to the android landing page comment: https://docs.mongodb.com/realm/sdk/android/#update-live-objects

**Pull request merged**